### PR TITLE
Fix for #3119

### PR DESCRIPTION
**Pull request by MattGertz**
@dotnet/roslyn-ide Please review.  

Note I considered factoring out the code common to If, If Directive, and ElseIf, but on consideration decided not to since (a) nothing else would ever need to use it and (b) inlining seemed better anyway.

### CODE REVIEW

File: src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/AddMissingTokensCodeCleanupProvider.vb [hunk 1]
```diff
@@ -508,16 +511,26 @@ Namespace Microsoft.CodeAnalysis.CodeCleanup.Providers
...
 
                         Else
-                            Dim ifDirective = TryCast(originalParent, IfDirectiveTriviaSyntax)
-                            If ifDirective IsNot Nothing Then
-                                Dim previousToken = ifDirective.Condition.GetLastToken(includeZeroWidth:=True)
-                                Dim nextToken = ifDirective.GetLastToken.GetNextToken
+                            Dim regularElseIf = TryCast(originalParent, ElseIfStatementSyntax)
+                            If regularElseIf IsNot Nothing Then
+                                Dim previousToken = regularElseIf.Condition.GetLastToken(includeZeroWidth:=True)
+                                Dim nextToken = regularElseIf.GetLastToken.GetNextToken
```
Comment (Pilchie) [inline src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/AddMissingTokensCodeCleanupProvider.vb]: I see that you're just doing the same thing that was already there, but why does `previousToken` use `includeZeroWidth`, but `nextToken` not?  @heejaechang any ideas?

**Comment by Pilchie** [comment]
:+1:, pending CLA signing and Jenkins completion :smile:

File: src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/AddMissingTokensCodeCleanupProvider.vb [hunk 2]
```diff
@@ -508,16 +511,26 @@ Namespace Microsoft.CodeAnalysis.CodeCleanup.Providers
...
                                 If Not InvalidOmittedToken(previousToken, nextToken) Then
-                                    Return ifDirective.WithThenKeyword(newToken)
+                                    Return regularElseIf.WithThenKeyword(newToken)
                                 End If
-                            End If
 
+                            Else
+                                Dim ifDirective = TryCast(originalParent, IfDirectiveTriviaSyntax)
+                                If ifDirective IsNot Nothing Then
+                                    Dim previousToken = ifDirective.Condition.GetLastToken(includeZeroWidth:=True)
```
Comment (jasonmalinowski) [inline src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/AddMissingTokensCodeCleanupProvider.vb]: Given this is the third repeat of this logic here, can we unify it a bit?

File: src/EditorFeatures/VisualBasicTest/LineCommit/CommitWithViewTests.vb [hunk 3]
```diff
@@ -1056,5 +1056,79 @@ End Module
                 Assert.Equal(expected.NormalizedValue, testData.Workspace.Documents.Single().TextBuffer.CurrentSnapshot.GetText())
             End Using
         End Sub
+
+        <WorkItem(3119, "https://github.com/dotnet/roslyn/issues/3119")>
+        <Fact, Trait(Traits.Feature, Traits.Features.LineCommit)>
+        Public Sub MissingThenInIf()
```
Comment (jasonmalinowski) [inline src/EditorFeatures/VisualBasicTest/LineCommit/CommitWithViewTests.vb]: Is there a reason this needs to be tested by the line commit tests as opposed to the direct tests for the code cleanup engine? I see there's AddMissingTokensTests.cs in the tree that might be more appropriate.

File: src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/AddMissingTokensCodeCleanupProvider.vb (hunk 1 above)
Comment (heejaechang) [inline src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/AddMissingTokensCodeCleanupProvider.vb]: I can't remember but probably due to EOL being attached to non-zero width token.

**Comment by jaredpar** [comment]
test unit32 please

**Pull request merged**